src/cm/nccl_ofi_cm_resources.cpp

```diff
@@ -152,11 +152,33 @@ cm_resources::cm_resources(nccl_net_ofi_domain_t &domain, size_t _conn_msg_data_
 
 cm_resources::~cm_resources()
 {
-	/* Resources can be destructed in the usual reverse-order, with one exception:
-	   The endpoint must be closed first, since posted buffers and requests cannot
-	   be freed until the endpoint is closed.
+	/* Libfabric imposes two key cleanup constraints:
+	 *
+	 * 1. Memory registrations (MRs) associated with an endpoint must be closed
+	 *    before the endpoint itself.
+	 *
+	 * 2. Memory buffers used with Libfabric transmission interfaces must not be
+	 *    released, deregistered, or reused until either a completion event is
+	 *    received or fi_close() on the endpoint returns. For receive buffers,
+	 *    this typically requires cancellation support, which not all providers
+	 *    implement.
+	 *
+	 * For non-FI_MR_ENDPOINT providers:
+	 *   We close the endpoint here before buffer cleanup. The first constraint
+	 *   does not apply (MRs are not bound to the endpoint), and closing the
+	 *   endpoint satisfies the second constraint.
+	 *
+	 * For FI_MR_ENDPOINT providers:
+	 *   We skip closing the endpoint here, as there are MRs in use. The remaining
```

Comment:
it's not that `there are MRs in use`.  It's that there are MRs still associated with this endpoint.